CalendarApp/GoogleCalendarService/Manager/EventsListBuilder.cs

```diff
@@ -0,0 +1,254 @@
+            {
+                case "SU":
+                    return DayOfWeek.Sunday;
+                case "MO":
+                    return DayOfWeek.Monday;
+                case "TU":
+                    return DayOfWeek.Tuesday;
+                case "WE":
+                    return DayOfWeek.Wednesday;
+                case "TH":
+                    return DayOfWeek.Thursday;
+                case "FR":
+                    return DayOfWeek.Friday;
+                case "SA":
+                    return DayOfWeek.Saturday;
+                default:
+                    return DayOfWeek.Sunday;
+            }
+        }
+
+        private bool CheckForCrossBreak(FrequencyType freqType, int i = 0, int count = 0, DateTime startTime = default(DateTime), DateTime untilDateTine = default(DateTime))
+        {
+            switch (freqType)
+            {
+                case FrequencyType.COUNT:
+                {
+                    if (i == count - 1)
+                        return true;
+                }
+                    break;
+
+                case FrequencyType.UNTIL:
+                {
+                    if(startTime.CompareTo(untilDateTine) > 0)
+                    return true;
+                }
+                    break;
+            }
+
+            return false;
+        }
+
+
+        private DateTime GetUntilDateTime(string untilDate)
+        {
+            string pattern = @"(?<Year>\d{4})(?<Month>\d{2})(?<Day>\d{2})T(?<Hour>\d{2})(?<Minute>\d{2})(?<Second>\d{2})Z";
+            Regex regex = new Regex(pattern);
+            Match match = regex.Match(untilDate);
+
+            var hour = match.Groups["Hour"].Value == "00" ? 0 : Convert.ToInt32(match.Groups["Hour"].Value);
+            var minute = match.Groups["Minute"].Value == "00" ? 0 : Convert.ToInt32(match.Groups["Minute"].Value);
+            var second = match.Groups["Second"].Value == "00" ? 0 : Convert.ToInt32(match.Groups["Second"].Value);
+
+            return new DateTime(
+                Convert.ToInt32(match.Groups["Year"].Value),
+                Convert.ToInt32(match.Groups["Month"].Value),
+                Convert.ToInt32(match.Groups["Day"].Value),
+                hour,
+                minute,
+                second
+                );
+        }
+
+        private List<int> GetDayJumpsList()
+        {
+            var resultList = new List<int>();
+            DateTime startTime = _realEvent.Start.DateTime.Value;
+
+            int i = 0;
+            if (_dayOfWeeksList.Count == 1)
+            {
+                while (true)
+                {
+                    resultList.Add(7);
+                    startTime = startTime.AddDays(7);
+                    if (CheckForCrossBreak(_freqType, i, _count, startTime, _untilDateTime))
+                        break;
+                    i++;
+                }
+            }
+            else
+            {
+                for(int j = 0, start = 0, target = 0;;i++, j++)
```

Comment:
I vote for giving i and j some understandable names. there is some logic behind those variables